src/core/lender/distributor/base.sol

```diff
@@ -108,9 +115,10 @@ contract Distributor is DSNote, Math {
     function repayTranches(uint available) public auth {
         // repay senior always first
         if(address(senior) != address(0)) {
-            if(junior.balance() > 0 && senior.debt() > 0) {
+            uint juniorBalance = currency.balanceOf(address(junior));
```

Comment:
Having the call twice here costs gas, we should try to avoid that.

Reply:
I have a PR open which fixes it.